const CustomError = require("../helpers/errorHelper");

const errorHandler = (err, req, res, next) => {
    let customError = err;

    if( err.name == "SyntaxError" ) {
        customError = new CustomError("Yaz覺m Hatas覺" +  err.message, 400);
    }
    if( err.message.includes("duplicate") ) {
        customError = new CustomError("kurs ad覺 kullan覺lmakta.", 400);
    }
   

    res.status(customError.status || 500)
    .json({
        success: false,
        message: customError.message
    })
}

module.exports = errorHandler;